src/components/utils/src/file_system.cc

```diff
@@ -197,76 +189,79 @@ void file_system::Close(std::ofstream* file_stream) {
 }
 
 std::string file_system::CurrentWorkingDirectory() {
-  const size_t filename_max_length = 1024;
-  char path[filename_max_length];
-  if (0 == getcwd(path, filename_max_length)) {
+  error_code ec;
+  fs::path currpath = fs::current_path(ec);
+  if (ec) {
     LOG4CXX_WARN(logger_, "Could not get CWD");
   }
-  return std::string(path);
+  return currpath.string();
 }
 
 std::string file_system::GetAbsolutePath(const std::string& path) {
-  char abs_path[PATH_MAX];
-  if (NULL == realpath(path.c_str(), abs_path)) {
-    return std::string();
+  error_code ec;
+  fs::path absolute = fs::canonical(path, ec);
+  if (ec) {
+    return std::string();  // invalid path
   }
-
-  return std::string(abs_path);
+  return absolute.string();
 }
 
 bool file_system::IsFileNameValid(const std::string& file_name) {
   return file_name.end() == std::find(file_name.begin(), file_name.end(), '/');
 }
 
+// Does not remove if file is write-protected
 bool file_system::DeleteFile(const std::string& name) {
   if (FileExists(name) && IsAccessible(name, W_OK)) {
-    return !remove(name.c_str());
+    return fs::remove(name.c_str());
```

Comment:
Need to use the error code variant, otherwise an exception will be thrown on error: https://www.boost.org/doc/libs/1_45_0/libs/filesystem/v3/doc/reference.html